lib/api/core/plugins/pluginContext.js

```diff
@@ -185,38 +204,41 @@ function execute (request, callback) {
  */
 function createUser(kuzzle, name, profile, userInfo) {
   const userRepository = kuzzle.repositories.user;
-
-  if (!userInfo) {
-    if (profile && typeof profile === 'object') {
-      userInfo = profile;
-      profile = null;
+  let
+    _profile = profile,
+    _userInfo = userInfo;
+
+  if (!_userInfo) {
+    if (_profile && typeof _profile === 'object') {
+      _userInfo = _profile;
+      _profile = null;
     }
     else {
-      userInfo = {};
+      _userInfo = {};
     }
   }
 
   if (!name || typeof name !== 'string') {
-    return Promise.reject(new PluginImplementationError('User creation error: "name" argument invalid or missing'));
+    return Bluebird.reject(new PluginImplementationError('User creation error: "name" argument invalid or missing'));
   }
 
-  if (typeof userInfo !== 'object' || Array.isArray(userInfo)) {
-    return Promise.reject(new PluginImplementationError('User creation error: invalid "info" argument (expected an object)'));
+  if (typeof _userInfo !== 'object' || Array.isArray(_userInfo)) {
+    return Bluebird.reject(new PluginImplementationError('User creation error: invalid "info" argument (expected an object)'));
   }
 
-  userInfo._id = name;
-  userInfo.profileIds = kuzzle.config.security.restrictedProfileIds;
-  if (profile) {
-    userInfo.profileIds = Array.isArray(profile) ? profile : [profile];
+  _userInfo._id = name;
+  _userInfo.profileIds = kuzzle.config.security.restrictedProfileIds;
+  if (_profile) {
+    _userInfo.profileIds = Array.isArray(_profile) ? _profile : [_profile];
   }
 
-  return userRepository.hydrate(new userRepository.ObjectConstructor(), userInfo)
+  return userRepository.hydrate(new userRepository.ObjectConstructor(), _userInfo)
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Split this 82 characters long line (which is greater than 80 authorized). [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3ALineLength)